### PR TITLE
Disable profiling when `getGraphExecutorOptimize` is unset

### PR DESCRIPTION
**Pull request by Krovatkin**
`getGraphExecutorOptimize` mandates we don't do any optimizations beyond what's required to run graphs. In this scenario, we don't want to do any profiling as profiling information will not be used.

### CODE REVIEW

**Review by ZolotukhinM — APPROVED** [review]
LGTM!

File: torch/csrc/jit/runtime/graph_executor.cpp [hunk 1]
```diff
@@ -736,7 +736,13 @@ c10::intrusive_ptr<Future> GraphExecutor::runAsync(Stack& stack) {
 }
 
 size_t GraphExecutor::getDefaultNumBailOuts() {
-  return getProfilingMode() ? getBailoutDepth().load() : 0;
```
Comment (eellison) [inline torch/csrc/jit/runtime/graph_executor.cpp:739]: Could we consider not even querying `GraphExecutor` if `! getGraphExecutorOptimize()` ?

Comment (eellison) [reply torch/csrc/jit/runtime/graph_executor.cpp:739]: Or dispatching to the SimpleExecutor if `!getGraphExecutorOptimize()`

Comment (eellison) [reply torch/csrc/jit/runtime/graph_executor.cpp:739]: From looking around maybe there's not a great place to do this... LGTM

Comment (ZolotukhinM) [reply torch/csrc/jit/runtime/graph_executor.cpp:739]: That was my impression as well tbh

File: torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp [hunk 2]
```diff
@@ -466,6 +447,15 @@ ExecutionPlan ProfilingGraphExecutorImpl::getPlanFor(
     return *optimized_plan_;
   }
 
+  // no opt mode
+  if (!getGraphExecutorOptimize()) {
+    auto copy = graph->copy();
```
Comment (eellison) [inline torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp:452]: This has the opposite problem now - if you reenable optimization after running it with disabling, you will never optimize.

**Review by eellison — COMMENTED** [review]
Why all the changes/

File: torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp (hunk 2 above)
Comment (Krovatkin) [reply torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp:452]: err i'm kinda thinking it should be okay. If the user changes the number of profiled runs midway, we won't update it for models in flight either.

File: torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp [hunk 3]
```diff
@@ -96,12 +96,7 @@ void runNooptPassPipeline(std::shared_ptr<Graph>& graph) {
   LowerGradOf(*graph);
   GRAPH_DEBUG("After LowerGradOf, before RemoveExpands\n", *graph);
   RemoveExpands(graph);
-  GRAPH_DEBUG("After RemoveExpands, before CanonicalizeOps\n", *graph);
```
Comment (eellison) [inline torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp:99]: Why was this changed as part of the PR?

**Review by eellison — COMMENTED** [review]
Are all of these changes intentional?

File: torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp (hunk 2 above)
Comment (Krovatkin) [reply torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp:452]: Alternatively, we could try disregarding `getGraphExecutorOptimize` we are already profiling but this seems complicated conceptually
